### PR TITLE
Handle OP_MSG with checksum

### LINKED ISSUE #26: Insert instruction exception
hi     Why is it that when I send 'db.person.insert({"name": "kelvin", "age": 35})' on the client side, the server parse throws an 'First OP_MSG payload type must be 0 not 1' exception

### PR DESCRIPTION
**Pull request by ajdavis**
Fixes #26.

### CODE REVIEW

File: mockupdb/__init__.py [hunk 1]
```diff
@@ -1631,7 +1666,7 @@ def _accept_loop(self):
                     server_thread.start()
             except socket.error as error:
                 if error.errno not in (
-                    errno.EAGAIN, errno.EBADF, errno.EWOULDBLOCK):
+                        errno.EAGAIN, errno.EBADF, errno.EWOULDBLOCK):
```
Comment (ajdavis) [inline mockupdb/__init__.py]: Oops, unrelated whitespace fix.

File: mockupdb/__init__.py [hunk 2]
```diff
@@ -275,7 +276,10 @@ def wait_until(predicate, success_description, timeout=10):
 
 OP_MSG_FLAGS = OrderedDict([
     ('checksumPresent', 1),
-    ('moreToCome', 2)])
+    ('moreToCome', 2),
+    ('exhaustAllowed', 16)])
```
Comment (ajdavis) [inline mockupdb/__init__.py]: Unrelated, adding the exhaustAllowed flag.

Comment (ShaneHarvey) [reply mockupdb/__init__.py]: Gotcha.
